src/scrapers/union-bank.js

```diff
@@ -0,0 +1,284 @@
+import {
+  dropdownSelect,
+  dropdownElements,
+  fillInput,
+  clickButton,
+  waitUntilElementFound,
+  pageEvalAll,
+  elementPresentOnPage,
+} from '../helpers/elements-interactions';
+import { waitForNavigation } from '../helpers/navigation';
+import { SHEKEL_CURRENCY, NORMAL_TXN_TYPE, TRANSACTION_STATUS } from '../constants';
+
+const BASE_URL = 'https://hb.unionbank.co.il';
+const TRANSACTIONS_URL = `${BASE_URL}/eBanking/Accounts/ExtendedActivity.aspx#/`;
+const DATE_FORMAT = 'DD/MM/YY';
+const NO_TRANSACTION_IN_DATE_RANGE_TEXT = 'לא קיימות תנועות מתאימות על פי הסינון שהוגדר';
+const DATE_HEADER = 'תאריך';
+const DESCRIPTION_HEADER = 'תיאור';
+const REFERENCE_HEADER = 'אסמכתא';
+const DEBIT_HEADER = 'חובה';
+const CREDIT_HEADER = 'זכות';
+const PENDING_TRANSACTIONS_TABLE_ID = 'trTodayActivityNapaTableUpper';
+const COMPLETED_TRANSACTIONS_TABLE_ID = 'ctlActivityTable';
+const ERROR_MESSAGE_CLASS = 'errInfo';
+const ACCOUNTS_DROPDOWN_SELECTOR = 'select#ddlAccounts_m_ddl';
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = [/eBanking\/Accounts/];
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = [/InternalSite\/CustomUpdate\/leumi\/LoginPage.ASP/];
+  return urls;
+}
+
+function createLoginFields(credentials) {
+  return [
+    { selector: '#uid', value: credentials.username },
+    { selector: '#password', value: credentials.password },
+  ];
+}
+
+function getAmountData(amountStr) {
+  const amountStrCopy = amountStr.replace(',', '');
+  return parseFloat(amountStrCopy);
+}
+
+function getTxnAmount(txn) {
+  const credit = getAmountData(txn.credit);
+  const debit = getAmountData(txn.debit);
+  return (Number.isNaN(credit) ? 0 : credit) - (Number.isNaN(debit) ? 0 : debit);
+}
+
+function convertTransactions(txns) {
+  return txns.map((txn) => {
+    const convertedDate = moment(txn.date, DATE_FORMAT).toISOString();
+    const convertedAmount = getTxnAmount(txn);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: txn.reference ? parseInt(txn.reference, 10) : null,
+      date: convertedDate,
+      processedDate: convertedDate,
+      originalAmount: convertedAmount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: convertedAmount,
+      status: txn.status,
+      description: txn.description,
+      memo: txn.memo,
+    };
+  });
+}
+
+function getTransactionDate(tds, txnsTableHeaders) {
+  return tds[txnsTableHeaders[DATE_HEADER]].trim();
+}
+
+function getTransactionDescription(tds, txnsTableHeaders) {
+  return tds[txnsTableHeaders[DESCRIPTION_HEADER]].trim();
+}
+
+function getTransactionReference(tds, txnsTableHeaders) {
+  return tds[txnsTableHeaders[REFERENCE_HEADER]].trim();
+}
+
+function getTransactionDebit(tds, txnsTableHeaders) {
+  return tds[txnsTableHeaders[DEBIT_HEADER]].trim();
+}
+
+function getTransactionCredit(tds, txnsTableHeaders) {
+  return tds[txnsTableHeaders[CREDIT_HEADER]].trim();
+}
```

Comment:
minor but might be relevant at runtime:
1. is it possible that `txnsTableHeaders[CREDIT_HEADER]` will result with null which will throw runtime error?
2. is it possible that `tds[txnsTableHeaders[CREDIT_HEADER]]` will result with null which will throw runtime error?

for the latter you can just add fallback value like `(tds[txnsTableHeaders[CREDIT_HEADER]] || '').trim()`. you can do the same on the first as well.